tests/src/Unit/Plugin/OgGroupResolver/OgRouteGroupResolverTestBase.php

```diff
@@ -0,0 +1,317 @@
+   */
+  protected $linkTemplates = [
+    'node' => [
+      'canonical' => '/node/{node}',
+      'delete-form' => '/node/{node}/delete',
+      'edit-form' => '/node/{node}/edit',
+      'version-history' => '/node/{node}/revisions',
+      'revision' => '/node/{node}/revisions/{node_revision}/view',
+    ],
+    'entity_test' => [
+      'canonical' => '/entity_test/{entity_test}',
+      'add-form' => '/entity_test/add',
+      'edit-form' => '/entity_test/manage/{entity_test}/edit',
+      'delete-form' => '/entity_test/delete/entity_test/{entity_test}',
+    ],
+    'taxonomy_term' => [
+      'canonical' => '/taxonomy/term/{taxonomy_term}',
+      'delete-form' => '/taxonomy/term/{taxonomy_term}/delete',
+      'edit-form' => '/taxonomy/term/{taxonomy_term}/edit',
+    ],
+  ];
+
+  /**
+   * The mocked route matcher.
+   *
+   * @var \Drupal\Core\Routing\RouteMatchInterface|\Prophecy\Prophecy\ObjectProphecy
+   */
+  protected $routeMatch;
+
+  /**
+   * The mocked OG group type manager.
+   *
+   * @var \Drupal\og\GroupTypeManager|\Prophecy\Prophecy\ObjectProphecy
+   */
+  protected $groupTypeManager;
+
+  /**
+   * The mocked entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface|\Prophecy\Prophecy\ObjectProphecy
+   */
+  protected $entityTypeManager;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    // Instantiate mocks of the classes that the plugins rely on.
+    $this->routeMatch = $this->prophesize(RouteMatchInterface::class);
+    $this->groupTypeManager = $this->prophesize(GroupTypeManager::class);
+    $this->entityTypeManager = $this->prophesize(EntityTypeManagerInterface::class);
+
+    // Create mocked test entities.
+    foreach ($this->getTestEntityProperties() as $id => $properties) {
+      $entity_type_id = $properties['type'];
+      $bundle_id = $properties['bundle'];
+      $is_group = !empty($properties['group']);
+      $is_group_content = !empty($properties['group_content']);
+
+      /** @var \Drupal\Core\Entity\ContentEntityInterface|\Prophecy\Prophecy\ObjectProphecy $entity */
+      $entity = $this->prophesize(ContentEntityInterface::class);
+      // In case this entity is questioned about its identity, it shall
+      // willingly pony up the requested information.
+      $entity->getEntityTypeId()->willReturn($properties['type']);
+      $entity->bundle()->willReturn($properties['bundle']);
+      $this->testEntities[$id] = $entity->reveal();
+
+      // It is not being tight lipped about whether it is a group or group
+      // content.
+      $this->groupTypeManager->isGroup($entity_type_id, $bundle_id)
+        ->willReturn($is_group);
+
+      // @todo Provide a mock for a call to ??::isGroupContent() as soon as this
+      //   is available in a service.
+      // $this->groupContentTypeManager??->isGroupContent($entity_type_id, $bundle_id)
+      //  ->willReturn($is_group_content);
```

Comment:
I'm using unit tests to test the plugins, but I hit a blocker: I need to be able to check if an entity is group content, but this is currently not possible in OO code. It is currently handled by `Og::isGroupContent()` which delegates to a static method `OgGroupAudienceHelper::getAllGroupAudienceFields()`. I cannot mock this :-1:

It seems like `OgGroupAudienceHelper` is one of the final bastions of procedural code left in OG. It only contains 2 methods so it will probably be a very quick conversion.

I will make a separate ticket for this.